cmake/sysbuild/sign_sysbuild.cmake

```diff
@@ -0,0 +1,326 @@
+  dt_reg_size(slot1_size TARGET mcuboot PATH ${slot1_path})
+  if(NOT write_block_size)
+    set(write_block_size 4)
+    message(WARNING "slot0_partition write block size devicetree parameter is missing, assuming write block size is 4")
+  endif()
+
+  # Fetch devicetree details for executable RAM region.
+  dt_chosen(ram_load_dev TARGET mcuboot PROPERTY "mcuboot,ram-load-dev")
+  if(DEFINED ram_load_dev)
+    dt_reg_addr(ram_load_address TARGET mcuboot PATH ${ram_load_dev})
+  else()
+    dt_chosen(chosen_ram TARGET mcuboot PROPERTY "zephyr,sram")
+    dt_reg_addr(ram_load_address TARGET mcuboot PATH ${chosen_ram})
+  endif()
+
+  # Fetch devicetree details for the active code partition.
+  dt_chosen(code_flash TARGET ${main_image} PROPERTY "zephyr,code-partition")
+  dt_reg_addr(code_addr TARGET ${main_image} PATH ${code_flash})
+  set(start_offset)
+  sysbuild_get(start_offset IMAGE ${main_image} VAR CONFIG_ROM_START_OFFSET KCONFIG)
+
+  # Append key file path.
+  if(NOT "${keyfile}" STREQUAL "")
+    set(imgtool_args --key "${keyfile}" ${imgtool_args})
+  endif()
+
+  # Construct imgtool command, based on the selected MCUboot mode.
+  set(imgtool_rom_command)
+  if(SB_CONFIG_MCUBOOT_MODE_DIRECT_XIP_WITH_REVERT OR SB_CONFIG_MCUBOOT_MODE_DIRECT_XIP)
+    if(is_secondary_app)
+      MATH(EXPR slot_offset "${code_addr} + ${start_offset} - ${slot1_addr}")
+      set(slot_size ${slot1_size})
+    else()
+      MATH(EXPR slot_offset "${code_addr} + ${start_offset} - ${slot0_addr}")
+      set(slot_size ${slot0_size})
+    endif()
+    set(imgtool_rom_command --rom-fixed ${code_addr})
+  elseif(SB_CONFIG_MCUBOOT_MODE_SINGLE_APP)
+      MATH(EXPR slot_offset "${code_addr} + ${start_offset} - ${slot0_addr}")
+      set(slot_size ${slot0_size})
+  elseif(SB_CONFIG_MCUBOOT_MODE_SINGLE_APP_RAM_LOAD)
+      MATH(EXPR slot_offset "${code_addr} + ${start_offset} - ${ram_load_address}")
+      set(slot_size ${slot0_size})
+      # RAM load requires setting the location of where to load the image to
+      set(imgtool_args --load-addr ${ram_load_address} ${imgtool_args})
+      set(write_block_size 1)
+  elseif(SB_CONFIG_MCUBOOT_MODE_RAM_LOAD)
+      MATH(EXPR slot_offset "${code_addr} + ${start_offset} - ${ram_load_address}")
+      set(slot_size ${slot1_size})
+      # RAM load requires setting the location of where to load the image to
+      set(imgtool_args --load-addr ${ram_load_address} ${imgtool_args})
+      set(write_block_size 1)
+      set(imgtool_args_alt_slot ${imgtool_args} --align 1 --hex-addr ${slot1_addr})
+      set(imgtool_args ${imgtool_args} --hex-addr ${slot0_addr})
+  elseif(SB_CONFIG_MCUBOOT_MODE_FIRMWARE_UPDATER)
+      if (CONFIG_MCUBOOT_APPLICATION_FIRMWARE_UPDATER)
+        MATH(EXPR slot_offset "${code_addr} + ${start_offset} - ${slot1_addr}")
+        set(slot_size ${slot1_size})
+      else()
+        MATH(EXPR slot_offset "${code_addr} + ${start_offset} - ${slot0_addr}")
+        set(slot_size ${slot0_size})
+      endif()
```

Comment:
since this is for one specific device only which won't be using these modes, these should go

Reply:
Currently we are trying all SWAP_MOVE, DIRECT_XIP, DIRECT_XIP_WITH_REVERT as well as FIRMWARE_UPDATER modes. I prefer to keep all of them for now.

Reply:
The call we had about design was **specifically** about direct-xip, there was no mention of ever using the other modes